spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java

```diff
@@ -83,7 +83,7 @@
 	/**
 	 * Maximum size, in bytes, of the HTTP message header.
 	 */
-	private int maxHttpHeaderSize = 0; // bytes
+	private int maxHttpHeaderSize = 8192;
```

Comment:
Please use `DataSize` and `DataSize.ofKiloBytes(8)` for the default value. The description should no longer mention "in bytes" since that's inferred by the type. The updated description and the value for that property should also be aligned to `8KB` in the appendix.

Reply:
@snicoll  thanks for review, two questions:
Q1: Do you mean change the type of maxHttpHeaderSize from int to DataSize ? That will need to do some refactor beccause there are some reference to getMaxHttpHeaderSize.  
Q2: How about other "in bytes" properties, do we also need to change their types to DataSize ? 
eg: https://github.com/spring-projects/spring-boot/blob/0590c4de3210a1cf41e63b928d46f99f6c0c5c6b/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java#L326
https://github.com/spring-projects/spring-boot/blob/0590c4de3210a1cf41e63b928d46f99f6c0c5c6b/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java#L331
https://github.com/spring-projects/spring-boot/blob/0590c4de3210a1cf41e63b928d46f99f6c0c5c6b/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java#L720
https://github.com/spring-projects/spring-boot/blob/0590c4de3210a1cf41e63b928d46f99f6c0c5c6b/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java#L933
https://github.com/spring-projects/spring-boot/blob/0590c4de3210a1cf41e63b928d46f99f6c0c5c6b/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java#L938

Reply:
> Do you mean change the type of maxHttpHeaderSize from int to DataSize ?

Yes

> How about other "in bytes" properties, do we also need to change their types to DataSize ?

Yes and we have a separate issue for that (we need to align their default value first and when we'll merge forward we'll do the change. I've added a note in #13821 (in any case this PR is about a specific task and should not be expanded to cover unrelated changes). Thanks!

Reply:
Thank you for explain, still have a question, if we make default value of maxHttpHeaderSize  to 8M, these code seem to make no sense.
https://github.com/spring-projects/spring-boot/blob/0590c4de3210a1cf41e63b928d46f99f6c0c5c6b/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/embedded/TomcatWebServerFactoryCustomizer.java#L117
I'm wondering why do you prefer `serverProperties.getMaxHttpHeaderSize()` than `serverProperties.getTomcat().getMaxHttpHeaderSize()` at the beginning?

Reply:
`8KB`, not `8MB`. The property must be deprecated and that method as well to offer a transition. If you don't know how to do this, I can take care of it in a polish commit.